src/target/riscv/riscv.h

```diff
@@ -326,17 +326,33 @@ struct riscv_info {
 
 	bool range_trigger_fallback_encountered;
 
-	bool riscv_ebreakm;
-	bool riscv_ebreaks;
-	bool riscv_ebreaku;
-
 	bool wp_allow_equality_match_trigger;
 	bool wp_allow_napot_trigger;
 	bool wp_allow_ge_lt_trigger;
 
 	bool autofence;
 };
 
+enum riscv_ebreak_ctl {
+	RISCV_DCSR_EBREAK_M_CTL,
+	RISCV_DCSR_EBREAK_S_CTL,
+	RISCV_DCSR_EBREAK_U_CTL,
+	RISCV_DCSR_EBREAK_VS_CTL,
+	RISCV_DCSR_EBREAK_VU_CTL,
+	N_RISCV_DCSR_EBREAK_CTL
+};
+
+struct riscv_private_config {
+	bool dcsr_ebreak_ctls[N_RISCV_DCSR_EBREAK_CTL];
```

Comment:
For clarity, I'd recommend to rename the field to `_bits` or `_fields` as opposed to `_ctls`.

(`ctrls` (controls) is a very general term that does not help code readers or reviewers much.)

```suggestion
	bool dcsr_ebreak_bits[N_RISCV_MODE];
```
or

```suggestion
	bool dcsr_ebreak_fields[N_RISCV_MODE];
```

Reply:
Addressed.